### DISCUSSION TITLE
Summary table

### DISCUSSION THREAD

**Discussion by Stakset** [fremtind/jokul#2520] [Annet]
Jeg lurer litt på den komponenten som heter "Summary table" i Jøkul.  Hvor mange er det som bruker den i dag? Føler den trenger litt kjærlighet både visuelt og funksjonelt. 
![Summary](https://user-images.githubusercontent.com/72978199/142627840-05692b89-d4f8-4a4c-b5c2-9a63315b493c.png)

Her er en nyere versjon fra PM. Mulighet for å sette tekst i fargen stein + mulighet for 2 linjer på "totalsum".

![Summary_PM](https://user-images.githubusercontent.com/72978199/142628336-7b505bf5-94e9-482f-a25a-a0bee8a80466.png)

**Comment by Stakset** [comment]
@karimarkhus @AudunBerdal Supert om dere legger inn eksempler fra FLYT og SAM :)

**Comment by Mikaila94** [comment]
Jeg var med på å skrive denne her, og slik den er lagt opp nå så tar den i mot en liste med objekter for å rendre tabellen.
Det kan hende det kan være lurt å ha litt mer fleksibilitet for denne, litt som det er gjort for Description List blant annet, at man heller tilbyr komponenter i form av `<thead>, <tbody>,  <tfooter>, <th>, <td>`.  Det kan jo vurderes om denne blir slått sammen med Tabell-komponenten hvis den er på vei til å forbedres.

**Comment by AudunBerdal** [comment]
![Frame 4177](https://user-images.githubusercontent.com/57794116/143596947-eb7b9123-6ce2-4e93-bbb1-ac50d6fc0df2.png)


Her er et eksempel på oppsummerings-tabellen vi jobber med på SAM

**Comment by onordbo** [comment]
Er ikke både `SummaryTable` og `DescriptionList` veldig like? Funksjonsmessig oppleves de slik i mitt hode, hvertfall. Har vi en mulighet for å smelte et snøfnugg her, @wkillerud? 🤔 Så lenge vi bygger inn litt ekstra fleksibilitet med støtte for ekstra kolonner og litt dividers, vel og merke. Eller begynner vi så smått å versere inn i mønster-land da?

**Reply by wkillerud** [reply]
For meg er det en nyanseforskjell, spesielt om det skal være støtte for flere kolonner. Da er vi i tabelland, men en [description list](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl) har aldri mer enn et _begrep+beskrivelse_-par.

Som @Mikaila94 er inne på kan det vurderes om Summary Table er et mønster for bruk av [Table](https://jokul.fremtind.no/komponenter/table), heller enn en egen komponent. Da må vi sørge for at vi har tabellkomponenter som klarer lage det visuelle som trengs.

**Reply by Stakset** [reply]
Hvordan tar vi dette videre da :)?

**Reply by wkillerud** [reply]
Kommer litt an på målet 😄 Vil vi fjerne SummaryTable som en egen komponent og bare bruke Table? Eller vil vi fremdeles skille på de to og oppdatere designet til SummaryTable?

For min del er det ikke så nøye – det blir mest opp til dere som bruker komponenten i dag. Tenker dere på det som "bare en vanlig tabell, og dette er tilfeldigvis innholdet"? Eller er den noe mer?

Kan @Mikaila94 @AudunBerdal og @Stakset ta en avgjørelse på det? ☝️ Gjerne på Teams, så oppsummere valget her i tråden.

Så blir aksjonspunktet etter det å:
- lage et Issue her på GitHub for å spore endringen
- lage en Figma-branch med endringer i Jøkul-biblioteket
- lage en Git-branch med kodeendringer og dokumentasjon
- merge begge brancher når alt er reviewet OK av en kollega :shipit:

**Comment by Mikaila94** [comment]
Har snakket med @Stakset og @AudunBerdal og om jeg ikke husker feil så kom vi fram til til at:

1. Vi beholder SummaryTable-komponenten, men forsøker å gjøre den mer fleksibel.  Vi ønsker å unngå å utvide Table til å bli en komponent som løser alle problemer. Table komponenten har annerledes styling og virker mer passende å bruke i situasjoner der man skal vise større mengder med data.  Lager en diskusjon for å diskutere implementasjonsdetaljer her [https://github.com/fremtind/jokul/issues/2617]

2. Både SAM og Flyt bruker en kombinasjon av Accordion + Table.  Dette kan enten bli implementert som egen komponent eller som utvidelse av Table (med ekspanderbare rader) eller muligens en kombinasjon av Table og SummaryTable. Eksapnderbare rader er noe vi ønsker på sikt ifølge dokumentasjonen til Table.  Føler ikke vi har kommet fram til noe på dette punktet, så trenger mer diskusjon på dette. Trenger en representant fra Flyt til å legge inn innspill til hvordan dere bruker tabeller her :  https://github.com/fremtind/jokul/issues/2618
